lib/rubygems/ext/cargo_builder.rb

```diff
@@ -4,26 +4,26 @@
 # over the `cargo rustc` command which takes care of building Rust code in a way
 # that Ruby can use.
 class Gem::Ext::CargoBuilder < Gem::Ext::Builder
-  attr_reader :spec
+  attr_reader :spec, :runner
 
-  def initialize(spec)
+  def initialize(spec, runner: nil)
     @spec = spec
+    @runner = runner || self.class.method(:run)
   end
 
   def build(_extension, dest_path, results, args = [], lib_dir = nil, cargo_dir = Dir.pwd)
     require "rubygems/command"
+    require_relative "cargo_builder/link_flag_converter"
     require "fileutils"
     require "shellwords"
 
-    build_crate(dest_path, results, args, cargo_dir)
+    build_crate(_extension, dest_path, results, args, cargo_dir)
     ext_path = rename_cdylib_for_ruby_compatibility(dest_path)
     finalize_directory(ext_path, dest_path, lib_dir, cargo_dir)
     results
   end
 
-  private
-
-  def build_crate(dest_path, results, args, cargo_dir)
+  def build_crate(_extension, dest_path, results, args, cargo_dir)
```

Comment:
I wasn't able to find out what's the reason for the `_extension` argument since it is not used. Can we remove it?